dace/transformation/subgraph/composite.py

```diff
@@ -72,7 +73,7 @@ def can_be_applied(self, sdfg: SDFG, subgraph: SubgraphView) -> bool:
             if expansion.can_be_applied(sdfg, subgraph):
                 # deepcopy
                 graph_indices = [i for (i,n) in enumerate(graph.nodes()) if n in subgraph]
-                sdfg_copy = SDFG.from_json(sdfg.to_json())
+                sdfg_copy = copy.deepcopy(sdfg)
```

Comment:
I think we did that to speed up the code, but given the new issue you opened it's safer to deepcopy